### PR TITLE
POWR-3908 Rework on BDS contacts

### PR DESCRIPTION
**Pull request by kxwang**

### CODE REVIEW

File: web/modules/custom/employees/employees.module [hunk 1]
```diff
@@ -604,3 +604,15 @@ function employees_preprocess_fieldset(&$variables) {
...
+/**
+ * Implement hook_preprocess_HOOK(&$variables) 
+ * Hide user email field if email starts with "no.email"
+ */
+function employees_preprocess_block(&$variables) {
+  if($variables["plugin_id"] == "field_block:user:user:mail" &&
+    substr($variables["content"][0]["#title"], 0, 8) == "no.email" ) {
+    $variables["content"] = [];
+  }
+}
```
Comment (joshuami) [inline web/modules/custom/employees/employees.module:618]: Needs a new line at the end of the file.

File: web/modules/custom/employees/employees.services.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+services:
+  employees.commands:
+    class: \Drupal\employees\Commands\BatchCommands
+    arguments: ['@entity_type.manager', '@logger.factory']
+    tags:
+      - { name: drush.command }
```
Comment (joshuami) [inline web/modules/custom/employees/employees.services.yml:6]: Needs a new line at the end of the file.

File: web/sites/default/config/core.extension.yml [hunk 3]
```diff
@@ -14,6 +14,7 @@ module:
   breakpoint: 0
   bulk_user_registration: 0
   ckeditor: 0
+  clamav: 0
```
Comment (joshuami) [inline web/sites/default/config/core.extension.yml:17]: This should not have been enabled locally I don't think. 🤔

File: web/themes/custom/cloudy2/templates/user/user--contact-details.html.twig [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        {% endif %}
+        {{ content.field_workspace }}
+      </div>
+      {# <div class="float-left align-left field__item">
+        {{ content.field_search_keywords }}
+      </div> #}
+    </div>
+  </div>
+  {% endblock %}
+</div>
```
Comment (joshuami) [inline web/themes/custom/cloudy2/templates/user/user--contact-details.html.twig:44]: Needs a new line at end of file.

**Review by joshuami — CHANGES_REQUESTED** [review]

File: web/modules/custom/employees/employees.module (hunk 1 above)
Comment (kxwang) [reply web/modules/custom/employees/employees.module]: Fixed

File: web/modules/custom/employees/employees.services.yml (hunk 2 above)
Comment (kxwang) [reply web/modules/custom/employees/employees.services.yml]: Fixed

File: web/sites/default/config/core.extension.yml (hunk 3 above)
Comment (kxwang) [reply web/sites/default/config/core.extension.yml]: Fixed

File: web/themes/custom/cloudy2/templates/user/user--contact-details.html.twig (hunk 4 above)
Comment (kxwang) [reply web/themes/custom/cloudy2/templates/user/user--contact-details.html.twig]: Fixed

File: web/sites/default/config/views.view.search_index.yml [hunk 5]
```diff
@@ -9,6 +9,8 @@ dependencies:
     - field.storage.node.field_p_card_allowed
     - field.storage.node.field_type
     - search_api.index.full_index
+    - system.menu.main
+    - user.role.authenticated
```
Comment (joshuami) [inline web/sites/default/config/views.view.search_index.yml:13]: @kxwang , we should not limit the contacts search to authenticated. Anonymous users can only access the site if they are on the city network and we want employees on the network to be able to look up this information.

File: web/sites/default/config/views.view.search_index.yml [hunk 6]
```diff
@@ -54,7 +56,7 @@ display:
       pager:
         type: full
         options:
-          items_per_page: 20
+          items_per_page: 50
```
Comment (joshuami) [inline web/sites/default/config/views.view.search_index.yml:59]: Why the bump to 50 results? That will make the page load a bit slower.

File: web/sites/default/config/views.view.search_index.yml [hunk 7]
```diff
@@ -84,12 +86,13 @@ display:
             'entity:media':
               document: search_result
             'entity:node':
+              contract: default
```
Comment (joshuami) [inline web/sites/default/config/views.view.search_index.yml:89]: Looks like contract was picked up in this view, we should set this to teaser.

**Review by joshuami — CHANGES_REQUESTED** [review]
@kxwang , I found a couple of small issues in the PR review. While you are in your local correcting the permission and display mode, could you rename the menu link from "Contacts" to "Directory".  We got some feedback from 311 yesterday that "directory" will make more sense to folks.

File: web/sites/default/config/views.view.search_index.yml (hunk 5 above)
Comment (kxwang) [reply web/sites/default/config/views.view.search_index.yml]: Removed

File: web/sites/default/config/views.view.search_index.yml (hunk 6 above)
Comment (kxwang) [reply web/sites/default/config/views.view.search_index.yml]: Reverted

File: web/sites/default/config/views.view.search_index.yml (hunk 7 above)
Comment (kxwang) [reply web/sites/default/config/views.view.search_index.yml]: done

**Pull request closed without merging**